### DISCUSSION TITLE
Feature request: MQTT integration

### DISCUSSION THREAD

**Discussion by fkromer** [eProsima/Integration-Service#147] [Ideas]
Would be great if one could integrate via [ROS2-Integration-Service](https://github.com/eProsima/ROS2-Integration-Service) and *soss* with MQTT. What do you guys think? From a high level view it should be possible to integrate *ROS2 topics <-> MQTT topics*.

**Comment by mxgrey** [comment]
Having an MQTT plugin would be great, but we're constrained by our available human resources and funding obligations, so I don't anticipate we'll be working on this at OSRC until it becomes a requirement for one of our contracts.

That being said, we'd be happy to review and merge a Pull Request that adds this feature. Alternatively it could be developed in its own repo, and we could just link to it from the `soss` README.

**Comment by fkromer** [comment]
Sure. I've created this issue because we will probably be interested in cooperation (e.g. funding based source code contribution). However if we can do that is in clarification ATM.

**Comment by fkromer** [comment]
@mxgrey We are terribly limited in human resources as well. The priorities of the company I'm working for has little interest in this topic ATM. From our side there is no need to keep this issue open.

**Comment by RFRIEDM-Trimble** [comment]
For anyone curious, [AWS Greengrass](https://aws.amazon.com/blogs/robotics/deploy-and-manage-ros-robots-with-aws-iot-greengrass-2-0-and-docker/) appears to have support for bridging ROS2 and MQTT together.
